src/main/scala/com/fulcrumgenomics/commons/io/CommandLineTool.scala

```diff
@@ -0,0 +1,216 @@
+      val process  = new ProcessBuilder(command: _*).redirectErrorStream(true).start()
+      val stdout   = ListBuffer[String]()
+      val sink     = new AsyncStreamSink(process.getInputStream, s => stdout.append(s))
+      val exitCode = process.waitFor()
+
+      require(exitCode == 0, s"Command did not execute successfully. Exit code: $exitCode")
+      yieldAndThen(stdout)(sink.safelyClose())
+    }
+  }
+
+  /** Returns true if the tool is available and false otherwise. */
+  lazy val Available: Boolean = execCommand(Executable +: TestArgs: _*).isSuccess
+}
+
+/** Indicates the executable can run scripts.*/
+trait ScriptRunner {
+  self: CommandLineTool =>
+
+  /** Suffix of scripts that can be run by this tool. */
+  val Suffix: FilenameSuffix
+
+  /** Executes a script from the classpath if this executable is available.
+    *
+    * @throws Exception when we are unable to execute to this script on the classpath with the given arguments.
+    * @param scriptResource the name of the script resource on the classpath
+    * @param args a variable list of arguments to pass to the script
+    */
+  def execIfAvailable(scriptResource: String, args: String*): Unit =
+    try { if (Available) exec(scriptResource, args: _*) }
+    catch { case e: Exception => throw new Exception(s"Cannot execute script from $scriptResource with args:$args") }
+
+  /** Executes a script stored at a Path if the this executable is available.
+    *
+    * @throws Exception when we are unable to execute to this script on the classpath with the given arguments.
+    * @param script Path of the script to be run
+    * @param args a variable list of arguments to pass to the script
+    */
+  def execIfAvailable(script: Path, args: String*): Unit = execIfAvailable(script.toString, args: _*)
+
+  /** Executes a script from the classpath, raise an exception otherwise.
+    *
+    * @throws ToolException when the exit code from the called process is not zero.
+    * @throws Exception when we are unable to execute to this script on the classpath with the given arguments.
+    * @param scriptResource the name of the script resource on the classpath
+    * @param args a variable list of arguments to pass to the script
+    */
+  def exec(scriptResource: String, args: String*): Unit = exec(writeResourceToTempFile(scriptResource), args: _*)
+
+  /** Executes a script from the filesystem Path.
+    *
+    * @throws ToolException when the exit code from the called process is not zero
+    * @throws Exception when we are unable to execute the script with the given arguments
+    * @param script Path to the script to be executed
+    * @param args a variable list of arguments to pass to the script
+    */
+  def exec(script: Path, args: String*): Unit = try {
+    val command = Executable +: script.toAbsolutePath.toString +: args
+    val process = new ProcessBuilder(command:_*).redirectErrorStream(false).start()
+    val pipe1   = Io.pipeStream(process.getErrorStream, logger.info)
```

Comment:
I think this should be:

```suggestion
    val pipe1   = Io.pipeStream(process.getErrorStream, logger.Warning)
```